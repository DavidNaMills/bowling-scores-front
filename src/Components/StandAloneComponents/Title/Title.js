import React from 'react';
import classes from './Title.module.scss';

const Title = ({ label = '', ttlType = 'sub' }) => {
    let item;

    switch (ttlType.toLowerCase()) {
        case 'main':
            item = <h1 className={classes.title__main}>{label}</h1>
            break;
        case 'section':
            item = <h2 className={classes.title__section}>{label}</h2>
            break;
        case 'sub':
            item = <h3 className={classes.title__sub}>{label}</h3>
            break;
        default:
            item = <h3 className={classes.title__sub}>{label}</h3>
            break;
    }

    return (
        <div className={classes.title}>
            {item}
        </div>
    )
}

export default Title;