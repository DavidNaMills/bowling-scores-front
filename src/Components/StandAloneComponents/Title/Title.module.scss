@import '../../../styles/colors.scss';
@import '../../../styles/fonts.scss';

.title{
    // -webkit-box-shadow: $title_shadow_full;
    // -moz-box-shadow: $title_shadow_full;
    // box-shadow: $title_shadow_full;
    padding: 2px;
    text-align: center;
    font-family: $header;

    &__main{
        font-size: $header_size_main;
        color: $title_main_color;
    }
    
    &__section{
        font-size: $header_size_section;
        color: $title_rest_color;
    }
    &__sub{
        font-size: $header_size_sub;
        color: $title_rest_color;
    }
}