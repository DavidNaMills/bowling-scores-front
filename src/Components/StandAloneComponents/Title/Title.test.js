import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import Title from './Title';

const testObject = [
    {
        msg: 'should display the correct message with type MAIN',
        snap: '<Title/> type MAIN snapshot test',
        theType: 'main',
        size: 'h1',
        className: 'title__main'
    },
    {
        msg: 'should display the correct message with type SECTION',
        snap: '<Title/> type SECTION snapshot test',
        theType: 'section',
        size: 'h2',
        className: 'title__section'
    },
    {
        msg: 'should display the correct message with type SUB',
        snap: '<Title/> type SUB snapshot test',
        theType: 'sub',
        size: 'h3',
        className: 'title__sub'
    },
    
]

describe('<Title/> component test suite', ()=>{
    const testLabel = 'Bowling Scores';

    describe('Title snapshot tests', ()=>{
        testObject.forEach(x=>{
            test(x.snap, ()=>{
                const component = renderer.create(<Title label={testLabel} ttleType={x.theType}/>);
                const app = component.toJSON();
                expect(app).toMatchSnapshot();
            })
        });
    })
    
    describe('Title styling tests', ()=>{
        testObject.forEach(x=>{
            it(x.msg, ()=>{
                const app = shallow(<Title label={testLabel} ttlType={x.theType}/>)
                const ttl = app.find(x.size);
                expect(ttl.length).toBe(1);
                expect(ttl.text()).toEqual(testLabel);
                expect(ttl.hasClass(x.className)).toBeTruthy();
            })
        });    
    })

})