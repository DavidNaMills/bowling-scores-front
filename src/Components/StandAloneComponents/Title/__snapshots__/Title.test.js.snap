// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title/> component test suite Title snapshot tests <Title/> type MAIN snapshot test 1`] = `
<div
  className="title"
>
  <h3
    className="title__sub"
  >
    Bowling Scores
  </h3>
</div>
`;

exports[`<Title/> component test suite Title snapshot tests <Title/> type SECTION snapshot test 1`] = `
<div
  className="title"
>
  <h3
    className="title__sub"
  >
    Bowling Scores
  </h3>
</div>
`;

exports[`<Title/> component test suite Title snapshot tests <Title/> type SUB snapshot test 1`] = `
<div
  className="title"
>
  <h3
    className="title__sub"
  >
    Bowling Scores
  </h3>
</div>
`;
